homeassistant/components/opentherm_gw/sensor.py

```diff
@@ -1,20 +1,621 @@
+        OpenThermSensorEntityDescription(
+            key=gw_vars.OTGW_CLOCKMHZ,
+            friendly_name_format="Gateway Clock Speed {}",
+        ),
+    ),
+    (
+        [gw_vars.OTGW],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.OTGW_LED_A,
+            friendly_name_format="Gateway LED A Mode {}",
+        ),
+    ),
+    (
+        [gw_vars.OTGW],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.OTGW_LED_B,
+            friendly_name_format="Gateway LED B Mode {}",
+        ),
+    ),
+    (
+        [gw_vars.OTGW],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.OTGW_LED_C,
+            friendly_name_format="Gateway LED C Mode {}",
+        ),
+    ),
+    (
+        [gw_vars.OTGW],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.OTGW_LED_D,
+            friendly_name_format="Gateway LED D Mode {}",
+        ),
+    ),
+    (
+        [gw_vars.OTGW],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.OTGW_LED_E,
+            friendly_name_format="Gateway LED E Mode {}",
+        ),
+    ),
+    (
+        [gw_vars.OTGW],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.OTGW_LED_F,
+            friendly_name_format="Gateway LED F Mode {}",
+        ),
+    ),
+    (
+        [gw_vars.OTGW],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.OTGW_GPIO_A,
+            friendly_name_format="Gateway GPIO A Mode {}",
+        ),
+    ),
+    (
+        [gw_vars.OTGW],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.OTGW_GPIO_B,
+            friendly_name_format="Gateway GPIO B Mode {}",
+        ),
+    ),
+    (
+        [gw_vars.OTGW],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.OTGW_SB_TEMP,
+            friendly_name_format="Gateway Setback Temperature {}",
+            device_class=SensorDeviceClass.TEMPERATURE,
+            state_class=SensorStateClass.MEASUREMENT,
+            native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+            suggested_display_precision=SENSOR_FLOAT_SUGGESTED_DISPLAY_PRECISION,
+        ),
+    ),
+    (
+        [gw_vars.OTGW],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.OTGW_SETP_OVRD_MODE,
+            friendly_name_format="Gateway Room Setpoint Override Mode {}",
+        ),
+    ),
+    (
+        [gw_vars.OTGW],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.OTGW_SMART_PWR,
+            friendly_name_format="Gateway Smart Power Mode {}",
+        ),
+    ),
+    (
+        [gw_vars.OTGW],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.OTGW_THRM_DETECT,
+            friendly_name_format="Gateway Thermostat Detection {}",
+        ),
+    ),
+    (
+        [gw_vars.OTGW],
+        OpenThermSensorEntityDescription(
+            key=gw_vars.OTGW_VREF,
+            friendly_name_format="Gateway Reference Voltage Setting {}",
+        ),
+    ),
```

Comment:
What does this return?

Reply:
Room Setpoint Override Mode: One of `T` (Temporary Override), `C` (Constant/Permanent override) or `N` (Override disabled)

Smart Power Mode: The active smart power mode (low/medium/high)

Thermostat Detection: The thermostat detection mode in use by the gateway. Some thermostats have quirks that need to be handled, but detection is not always reliable. The active mode can be changed between `C` (force Remeha Celsia 20), `I` (force Remeha iSense), `S` (force standard thermostat) or `D` (auto-detect).

Reference voltage setting: The gateway has an adjustable reference voltage that is used to receive protocol messages. This reports that setting in a range between `0` and `9`, where actual voltages depend on the PIC used in the gateway.